core/sql/parser/sqlparser.y

```diff
@@ -7001,6 +7036,46 @@ rel_subquery_and_as_clause : rel_subquery as_clause
                                $$ = $2;
                              }
 
+with_clause_list : with_clause
+                     {  $$ = $1 ; }
+
+                   | with_clause_list ',' correlation_name TOK_AS rel_subquery 
```

Comment:
@zellerh , I cannot figure out how to do this. Could you please give me an example? I am very new to the parser.